### PR TITLE
[assimp] Fix generated -config files to use more generic form

### LINKED ISSUE #9022: assimp build failure using CMake
Problem seems to be related to linking of irrXML and ZLIB dependencies.

**Host Environment**
- OS: Windows
- Compiler: MSVC 19.23.28106.4
- CMake generator: Ninja

**Failure logs** 
```
[build]    Creating library assimp-test.lib and object assimp-test.exp
[build] assimp-vc142-mtd.lib(D3MFOpcPackage.cpp.obj) : error LNK2001: unresolved external symbol "class irr::io::IIrrXMLReader<char,class irr::io::IXMLBase> * __cdecl irr::io::createIrrXMLReader(class irr::io::IFileReadCallBack *)" (?createIrrXMLReader@io@irr@@YAPEAV?$IIrrXMLReader@DVIXMLBase@io@irr@@@12@PEAVIFileReadCallBack@12@@Z)
[build] assimp-vc142-mtd.lib(FIReader.cpp.obj) : error LNK2001: unresolved external symbol "class irr::io::IIrrXMLReader<char,class irr::io::IXMLBase> * __cdecl irr::io::createIrrXMLReader(class irr::io::IFileReadCallBack *)" (?createIrrXMLReader@io@irr@@YAPEAV?$IIrrXMLReader@DVIXMLBase@io@irr@@@12@PEAVIFileReadCallBack@12@@Z)
[build] assimp-vc142-mtd.lib(XGLLoader.cpp.obj) : error LNK2001: unresolved external symbol "class irr::io::IIrrXMLReader<char,class irr::io::IXMLBase> * __cdecl irr::io::createIrrXMLReader(class irr::io::IFileReadCallBack *)" (?createIrrXMLReader@io@irr@@YAPEAV?$IIrrXMLReader@DVIXMLBase@io@irr@@@12@PEAVIFileReadCallBack@12@@Z)
[build] assimp-vc142-mtd.lib(D3MFImporter.cpp.obj) : error LNK2001: unresolved external symbol "class irr::io::IIrrXMLReader<char,class irr::io::IXMLBase> * __cdecl irr::io::createIrrXMLReader(class irr::io::IFileReadCallBack *)" (?createIrrXMLReader@io@irr@@YAPEAV?$IIrrXMLReader@DVIXMLBase@io@irr@@@12@PEAVIFileReadCallBack@12@@Z)
[build] assimp-vc142-mtd.lib(ColladaParser.cpp.obj) : error LNK2001: unresolved external symbol "class irr::io::IIrrXMLReader<char,class irr::io::IXMLBase> * __cdecl irr::io::createIrrXMLReader(class irr::io::IFileReadCallBack *)" (?createIrrXMLReader@io@irr@@YAPEAV?$IIrrXMLReader@DVIXMLBase@io@irr@@@12@PEAVIFileReadCallBack@12@@Z)
[build] assimp-vc142-mtd.lib(OgreXmlSerializer.cpp.obj) : error LNK2001: unresolved external symbol "class irr::io::IIrrXMLReader<char,class irr::io::IXMLBase> * __cdecl irr::io::createIrrXMLReader(class irr::io::IFileReadCallBack *)" (?createIrrXMLReader@io@irr@@YAPEAV?$IIrrXMLReader@DVIXMLBase@io@irr@@@12@PEAVIFileReadCallBack@12@@Z)
[build] assimp-vc142-mtd.lib(AMFImporter.cpp.obj) : error LNK2001: unresolved external symbol "class irr::io::IIrrXMLReader<char,class irr::io::IXMLBase> * __cdecl irr::io::createIrrXMLReader(class irr::io::IFileReadCallBack *)" (?createIrrXMLReader@io@irr@@YAPEAV?$IIrrXMLReader@DVIXMLBase@io@irr@@@12@PEAVIFileReadCallBack@12@@Z)
[build] assimp-vc142-mtd.lib(IRRMeshLoader.cpp.obj) : error LNK2001: unresolved external symbol "class irr::io::IIrrXMLReader<char,class irr::io::IXMLBase> * __cdecl irr::io::createIrrXMLReader(class irr::io::IFileReadCallBack *)" (?createIrrXMLReader@io@irr@@YAPEAV?$IIrrXMLReader@DVIXMLBase@io@irr@@@12@PEAVIFileReadCallBack@12@@Z)
[build] assimp-vc142-mtd.lib(IRRLoader.cpp.obj) : error LNK2001: unresolved external symbol "class irr::io::IIrrXMLReader<char,class irr::io::IXMLBase> * __cdecl irr::io::createIrrXMLReader(class irr::io::IFileReadCallBack *)" (?createIrrXMLReader@io@irr@@YAPEAV?$IIrrXMLReader@DVIXMLBase@io@irr@@@12@PEAVIFileReadCallBack@12@@Z)
[build] assimp-vc142-mtd.lib(OgreImporter.cpp.obj) : error LNK2001: unresolved external symbol "class irr::io::IIrrXMLReader<char,class irr::io::IXMLBase> * __cdecl irr::io::createIrrXMLReader(class irr::io::IFileReadCallBack *)" (?createIrrXMLReader@io@irr@@YAPEAV?$IIrrXMLReader@DVIXMLBase@io@irr@@@12@PEAVIFileReadCallBack@12@@Z)
[build] assimp-vc142-mtd.lib(FBXParser.cpp.obj) : error LNK2001: unresolved external symbol inflate
[build] assimp-vc142-mtd.lib(BlenderLoader.cpp.obj) : error LNK2001: unresolved external symbol inflate
[build] assimp-vc142-mtd.lib(XGLLoader.cpp.obj) : error LNK2001: unresolved external symbol inflate
[build] assimp-vc142-mtd.lib(XFileParser.cpp.obj) : error LNK2001: unresolved external symbol inflate
[build] assimp-vc142-mtd.lib(unzip.c.obj) : error LNK2001: unresolved external symbol inflate
[build] assimp-vc142-mtd.lib(FBXParser.cpp.obj) : error LNK2001: unresolved external symbol inflateEnd
[build] assimp-vc142-mtd.lib(BlenderLoader.cpp.obj) : error LNK2001: unresolved external symbol inflateEnd
[build] assimp-vc142-mtd.lib(XGLLoader.cpp.obj) : error LNK2001: unresolved external symbol inflateEnd
[build] assimp-vc142-mtd.lib(XFileParser.cpp.obj) : error LNK2001: unresolved external symbol inflateEnd
[build] assimp-vc142-mtd.lib(unzip.c.obj) : error LNK2001: unresolved external symbol inflateEnd
[build] assimp-vc142-mtd.lib(BlenderLoader.cpp.obj) : error LNK2019: unresolved external symbol inflateInit2_ referenced in function "protected: virtual void __cdecl Assimp::BlenderImporter::InternReadFile(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,struct aiScene *,class Assimp::IOSystem *)" (?InternReadFile@BlenderImporter@Assimp@@MEAAXAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@PEAUaiScene@@PEAVIOSystem@2@@Z)
[build] assimp-vc142-mtd.lib(XGLLoader.cpp.obj) : error LNK2001: unresolved external symbol inflateInit2_
[build] assimp-vc142-mtd.lib(XFileParser.cpp.obj) : error LNK2001: unresolved external symbol inflateInit2_
[build] assimp-vc142-mtd.lib(unzip.c.obj) : error LNK2001: unresolved external symbol inflateInit2_
[build] assimp-vc142-mtd.lib(AssbinLoader.cpp.obj) : error LNK2019: unresolved external symbol uncompress referenced in function "public: virtual void __cdecl Assimp::AssbinImporter::InternReadFile(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,struct aiScene *,class Assimp::IOSystem *)" (?InternReadFile@AssbinImporter@Assimp@@UEAAXAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@PEAUaiScene@@PEAVIOSystem@2@@Z)
[build] assimp-vc142-mtd.lib(XFileParser.cpp.obj) : error LNK2019: unresolved external symbol inflateSetDictionary referenced in function "public: __cdecl Assimp::XFileParser::XFileParser(class std::vector<char,class std::allocator<char> > const &)" (??0XFileParser@Assimp@@QEAA@AEBV?$vector@DV?$allocator@D@std@@@std@@@Z)
[build] assimp-vc142-mtd.lib(XFileParser.cpp.obj) : error LNK2019: unresolved external symbol inflateReset referenced in function "public: __cdecl Assimp::XFileParser::XFileParser(class std::vector<char,class std::allocator<char> > const &)" (??0XFileParser@Assimp@@QEAA@AEBV?$vector@DV?$allocator@D@std@@@std@@@Z)
[build] assimp-vc142-mtd.lib(unzip.c.obj) : error LNK2019: unresolved external symbol crc32 referenced in function unzReadCurrentFile
[build] assimp-vc142-mtd.lib(unzip.c.obj) : error LNK2019: unresolved external symbol get_crc_table referenced in function unzOpenCurrentFile3
[build] assimp-vc142-mtd.lib(FBXParser.cpp.obj) : error LNK2019: unresolved external symbol inflateInit_ referenced in function "void __cdecl Assimp::FBX::`anonymous namespace'::ReadBinaryDataArray(char,unsigned int,char const * &,char const *,class std::vector<char,class std::allocator<char> > &,class Assimp::FBX::Element const &)" (?ReadBinaryDataArray@?A0x7dac45b3@FBX@Assimp@@YAXDIAEAPEBDPEBDAEAV?$vector@DV?$allocator@D@std@@@std@@AEBVElement@23@@Z)
[build] assimp-test.exe : fatal error LNK1120: 10 unresolved externals
[build] ninja: build stopped: subcommand failed.
[build] Build finished with exit code 1
```

**Additional context**
Triplet being used is assimp:x64-windows-static

**CMakeLists.txt**
```
cmake_minimum_required(VERSION 3.15)
set(CMAKE_CXX_STANDARD 17)

set(VCPKG_ROOT ${CMAKE_SOURCE_DIR}/vcpkg)

set(CMAKE_TOOLCHAIN_FILE ${VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake)
set(CMAKE_PREFIX_PATH ${VCPKG_ROOT}/installed/${VCPKG_TARGET_TRIPLET})

project(assimp-test)

find_package(assimp CONFIG REQUIRED)

add_executable(assimp-test src/main.cpp)
set_property(TARGET assimp-test PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreadedDebug")
target_link_libraries(assimp-test PRIVATE assimp::assimp)
```

**main.cpp**
```
#include <assimp/Importer.hpp>

int main()
{
  Assimp::Importer import;
}
```

### LINKED ISSUE #9918: assimp package does not link irrXML
**Describe the bug**
I have a simple program that uses `assimp`.

CMakeLists.txt is configured with the following:

```
find_package(assimp CONFIG REQUIRED)
target_link_libraries(cubist PRIVATE assimp::assimp)
#target_link_libraries(cubist PRIVATE /home/chrome/code/vcpkg/installed/x64-linux/lib/libIrrXML.a)
```

Program does not build unless I comment out the second `target_link_libraries` line.

**Environment**
- OS: linux
- Compiler: clang version 9.0.0-2 (tags/RELEASE_900/final)

**To Reproduce**
1. ./vcpkg install assimp
2. Add assimp to your project and include the files.
3. Try to build

**Expected behavior**
Expect not to see compiler errors.

**Failure logs** 

```
[build] [5/5 100% :: 1.151] Linking CXX executable cubist
[build] FAILED: cubist 
[build] : && /bin/clang++-9  -g   CMakeFiles/cubist.dir/src/Mesh.cpp.o CMakeFiles/cubist.dir/src/Model.cpp.o CMakeFiles/cubist.dir/src/ShaderProgram.cpp.o CMakeFiles/cubist.dir/src/main.cpp.o  -o cubist  /home/chrome/code/vcpkg/installed/x64-linux/debug/lib/libspdlogd.a /home/chrome/code/vcpkg/installed/x64-linux/debug/lib/libglad.a /home/chrome/code/vcpkg/installed/x64-linux/debug/lib/libglfw3.a /home/chrome/code/vcpkg/installed/x64-linux/debug/lib/libassimpd.a /home/chrome/code/vcpkg/installed/x64-linux/debug/lib/libz.a /home/chrome/code/vcpkg/installed/x64-linux/debug/lib/libfmtd.a /usr/lib/x86_64-linux-gnu/librt.so -lm -ldl /usr/lib/x86_64-linux-gnu/libX11.so -lpthread && :
[build] ld: error: undefined symbol: irr::io::createIrrXMLReader(irr::io::IFileReadCallBack*)
[build] >>> referenced by AMFImporter.cpp:414 (/home/chrome/code/vcpkg/buildtrees/assimp/src/v5.0.0-fa91f462f7/code/AMF/AMFImporter.cpp:414)
[build] >>>               AMFImporter.cpp.o:(Assimp::AMFImporter::ParseFile(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, Assimp::IOSystem*)) in archive /home/chrome/code/vcpkg/installed/x64-linux/debug/lib/libassimpd.a
[build] >>> referenced by ColladaParser.cpp:125 (/home/chrome/code/vcpkg/buildtrees/assimp/src/v5.0.0-fa91f462f7/code/Collada/ColladaParser.cpp:125)
[build] >>>               ColladaParser.cpp.o:(Assimp::ColladaParser::ColladaParser(Assimp::IOSystem*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)) in archive /home/chrome/code/vcpkg/installed/x64-linux/debug/lib/libassimpd.a
[build] >>> referenced by ColladaParser.cpp:168 (/home/chrome/code/vcpkg/buildtrees/assimp/src/v5.0.0-fa91f462f7/code/Collada/ColladaParser.cpp:168)
[build] >>>               ColladaParser.cpp.o:(Assimp::ColladaParser::ReadZaeManifest[abi:cxx11](Assimp::ZipArchiveIOSystem&)) in archive /home/chrome/code/vcpkg/installed/x64-linux/debug/lib/libassimpd.a
[build] >>> referenced by IRRMeshLoader.cpp:148 (/home/chrome/code/vcpkg/buildtrees/assimp/src/v5.0.0-fa91f462f7/code/Irr/IRRMeshLoader.cpp:148)
[build] >>>               IRRMeshLoader.cpp.o:(Assimp::IRRMeshImporter::InternReadFile(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, aiScene*, Assimp::IOSystem*)) in archive /home/chrome/code/vcpkg/installed/x64-linux/debug/lib/libassimpd.a
[build] >>> referenced by IRRLoader.cpp:915 (/home/chrome/code/vcpkg/buildtrees/assimp/src/v5.0.0-fa91f462f7/code/Irr/IRRLoader.cpp:915)
[build] >>>               IRRLoader.cpp.o:(Assimp::IRRImporter::InternReadFile(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, aiScene*, Assimp::IOSystem*)) in archive /home/chrome/code/vcpkg/installed/x64-linux/debug/lib/libassimpd.a
[build] >>> referenced by OgreImporter.cpp:131 (/home/chrome/code/vcpkg/buildtrees/assimp/src/v5.0.0-fa91f462f7/code/Ogre/OgreImporter.cpp:131)
[build] >>>               OgreImporter.cpp.o:(Assimp::Ogre::OgreImporter::InternReadFile(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, aiScene*, Assimp::IOSystem*)) in archive /home/chrome/code/vcpkg/installed/x64-linux/debug/lib/libassimpd.a
[build] >>> referenced by OgreXmlSerializer.cpp:744 (/home/chrome/code/vcpkg/buildtrees/assimp/src/v5.0.0-fa91f462f7/code/Ogre/OgreXmlSerializer.cpp:744)
[build] >>>               OgreXmlSerializer.cpp.o:(Assimp::Ogre::OgreXmlSerializer::OpenReader(Assimp::IOSystem*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)) in archive /home/chrome/code/vcpkg/installed/x64-linux/debug/lib/libassimpd.a
[build] >>> referenced by XGLLoader.cpp:213 (/home/chrome/code/vcpkg/buildtrees/assimp/src/v5.0.0-fa91f462f7/code/XGL/XGLLoader.cpp:213)
[build] >>>               XGLLoader.cpp.o:(Assimp::XGLImporter::InternReadFile(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, aiScene*, Assimp::IOSystem*)) in archive /home/chrome/code/vcpkg/installed/x64-linux/debug/lib/libassimpd.a
[build] >>> referenced by FIReader.cpp:1832 (/home/chrome/code/vcpkg/buildtrees/assimp/src/v5.0.0-fa91f462f7/code/X3D/FIReader.cpp:1832)
[build] >>>               FIReader.cpp.o:(Assimp::FIReader::create(Assimp::IOStream*)) in archive /home/chrome/code/vcpkg/installed/x64-linux/debug/lib/libassimpd.a
[build] >>> referenced by D3MFImporter.cpp:474 (/home/chrome/code/vcpkg/buildtrees/assimp/src/v5.0.0-fa91f462f7/code/3MF/D3MFImporter.cpp:474)
[build] >>>               D3MFImporter.cpp.o:(Assimp::D3MFImporter::InternReadFile(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, aiScene*, Assimp::IOSystem*)) in archive /home/chrome/code/vcpkg/installed/x64-linux/debug/lib/libassimpd.a
[build] >>> referenced 1 more times
```

### LINKED ISSUE #12154: assimp build failure
**Describe the bug**
Projects with assimp dependencies do not build (without workarounds).

**Environment**
- OS: Linux
- Compiler: g++ 9.3

**To Reproduce**
1. ./vcpkg install assimp
Post-installation instructions state:
```
The package assimp:x64-linux provides CMake targets:

    find_package(assimp CONFIG REQUIRED)
    target_link_libraries(main PRIVATE assimp::assimp)
```

2. Simple main.cpp with assimp dependency:
```
#include <assimp/Importer.hpp>

int main()
{
    Assimp::Importer importer;
    return 0;
}
```

3: CMakeLists.txt:
```
cmake_minimum_required(VERSION 3.15)

if(DEFINED ENV{VCPKG_ROOT} AND NOT DEFINED CMAKE_TOOLCHAIN_FILE)
  set(CMAKE_TOOLCHAIN_FILE "$ENV{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake" CACHE STRING "")
endif()

project (AssimpTest)
set(CMAKE_CXX_STANDARD 17)
add_compile_options(-Wall -Wextra -pedantic -Werror -Wno-ignored-qualifiers)

set(TARGET_NAME AssimpTest)
find_package(assimp CONFIG REQUIRED)
add_executable(${TARGET_NAME} "main.cpp")
target_link_libraries(${TARGET_NAME} PRIVATE assimp::assimp)
```

4. cmake .
5. make .

**Expected behavior**
A successful build that properly identifies and incorporates include paths and linker paths for assimp and assimp dependencies.

**Failure logs** 
CMakeFiles/AssimpTest.dir/build.make:82: *** target pattern contains no '%'.  Stop.
make[1]: *** [CMakeFiles/Makefile2:75: CMakeFiles/AssimpTest.dir/all] Error 2
make: *** [Makefile:84: all] Error 2

CMakeFiles/AssimpTest.dir/build.make line 82 contains:
AssimpTest: assimp::assimp-NOTFOUND

**Additional context**
If you were to replace assimp::assimp with simply assimp (which is the value of ${ASSIMP_LIBRARIES} -- the target used in the previous port) and rebuild you then get a compiler error:
main.cpp:1:10: fatal error: assimp/Importer.hpp: No such file or directory
    1 | #include <assimp/Importer.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~
compilation terminated.

i.e. the include directory is not set for this dependency. This can be addressed by adding the following to CMakeLists.txt (before the add_executable command):
```
set(INCLUDE_DIR "$ENV{VCPKG_ROOT}/installed/${VCPKG_TARGET_TRIPLET}/include")
include_directories(${INCLUDE_DIR})
```

This workaround allows the compilation to succeed but linking now fails with:
/usr/bin/ld: cannot find -lassimp
collect2: error: ld returned 1 exit status

i.e. the library directory is not set for this dependency. We can add:
```
set(LIBRARY_DIR "$ENV{VCPKG_ROOT}/installed/${VCPKG_TARGET_TRIPLET}/lib")
link_directories(${LIBRARY_DIR})
```

And now the linker will find libassimp.a. However, you'll now see numerous linker errors related to two libraries that assimp depends upon: libz.a and libIrrXML.a. Both of these libraries are installed with the installation of the assimp package but they are not included with a target_link_libraries(${TARGET_NAME} PRIVATE assimp::assimp). Instead, you must manually specify these dependencies by adding:
```
target_link_libraries(${TARGET_NAME} PRIVATE z)
target_link_libraries(${TARGET_NAME} PRIVATE IrrXML)
```

Thus, a final CMakeLists.txt file would be:
```
cmake_minimum_required(VERSION 3.15)

if(DEFINED ENV{VCPKG_ROOT} AND NOT DEFINED CMAKE_TOOLCHAIN_FILE)
  set(CMAKE_TOOLCHAIN_FILE "$ENV{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake" CACHE STRING "")
endif()

project (AssimpTest)

set(CMAKE_CXX_STANDARD 17)
add_compile_options(-Wall -Wextra -pedantic -Werror -Wno-ignored-qualifiers)

set(TARGET_NAME AssimpTest)

find_package(assimp CONFIG REQUIRED)

set(INCLUDE_DIR "$ENV{VCPKG_ROOT}/installed/${VCPKG_TARGET_TRIPLET}/include")
set(LIBRARY_DIR "$ENV{VCPKG_ROOT}/installed/${VCPKG_TARGET_TRIPLET}/lib")
include_directories(${INCLUDE_DIR})
link_directories(${LIBRARY_DIR})
add_executable(${TARGET_NAME} "main.cpp")

target_link_libraries(${TARGET_NAME} PRIVATE assimp)
target_link_libraries(${TARGET_NAME} PRIVATE z)
target_link_libraries(${TARGET_NAME} PRIVATE IrrXML)
```

This produces a successfully linked binary. However, none of that should be necessary -- and did not exist in a previous revision of this package. I see numerous commits to https://github.com/microsoft/vcpkg/tree/master/ports/assimp between 5 and 9 months ago. I suspect the error was introduced somewhere in that history.

Looking at existing issues and I see similar reports in a [number of open issues](https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+assimp), but I see no resolution.

### LINKED ISSUE #13253: [assimp] static link failure
The current port file applies a bunch of replacements to the assimp-config.cmake, problem is that assimp-config.cmake only has a include “include(${CMAKE_CURRENT_LIST_DIR}/assimpTargets.cmake)” hence the fixes does nothing

seems the port file was broken by this commit
https://github.com/microsoft/vcpkg/commit/94b7f9a6a8c527d7dbdb3b78a46f0874b79712a4#diff-d78ed4342bc7afa31f0e0d467408e094

in assimp version 4.1.0 there was not CMake targets defined. But in version 5.0.0 assimp defines targets hence the replacements no longer work and the fixes has to be updated.

### PR DESCRIPTION
**Pull request by ras0219**
Assimp has two sets of config files: one that properly handles dependencies and one that doesn't. This PR switches which set to install.

Additionally, this PR causes the embedded IrrXML source to be built directly into the assimp library instead of as a separate binary for simpler consumption.

- What does your PR fix? Fixes #
Fixes #13253 #12154 #9918 #9022

### CODE REVIEW

File: ports/assimp/portfile.cmake [hunk 1]
```diff
@@ -17,53 +18,27 @@ file(REMOVE_RECURSE ${SOURCE_PATH}/contrib/zlib ${SOURCE_PATH}/contrib/gtest ${S
...
-  set(VCPKG_BUILD_SHARED_LIBS OFF)
-endif()
-
 vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
     PREFER_NINJA
     OPTIONS -DASSIMP_BUILD_TESTS=OFF
             -DASSIMP_BUILD_ASSIMP_VIEW=OFF
             -DASSIMP_BUILD_ZLIB=OFF
-            -DASSIMP_BUILD_SHARED_LIBS=${VCPKG_BUILD_SHARED_LIBS}
```
Comment (BillyONeal) [inline ports/assimp/portfile.cmake:32]: Does assimp populate this value from `BUILD_SHARED_LIBS`?

**Comment by BillyONeal** [comment]
How does this compare to #12437 ?

File: ports/assimp/portfile.cmake (hunk 1 above)
Comment (ras0219-msft) [reply ports/assimp/portfile.cmake:32]: Yep

**Comment by BillyONeal** [comment]
OK, this is fine to me but please sync with @JackBoosY on the other review.

**Comment by NancyLi1013** [comment]
The failures on x64-linux due to this:

```
CMake Error at scripts/cmake/vcpkg_fixup_cmake_targets.cmake:72 (message):
  '/mnt/vcpkg-ci/packages/assimp_x64-linux/debug/lib/cmake/assimp' does not
  exist.
Call Stack (most recent call first):
  ports/assimp/portfile.cmake:37 (vcpkg_fixup_cmake_targets)
  scripts/ports.cmake:79 (include)
```

**Comment by petersteneteg** [comment]
Assimps own hardcoded config files uses the target name "assimp::assimp" but with this change the name will be "Assimp::assimp" which breaks compatibility...

**Comment by ras0219** [comment]
I've added in a backwards-compatible alias that forwards `assimp::assimp` to `Assimp::assimp`. It's unfortunate that their cmake files have two different target names, but that's life I suppose.

**Comment by cenit** [comment]
In #12199 I had to modify assimp a lot, involving some of the things here

**Comment by JackBoosY** [comment]
Will test usage on Linux.

**Comment by JackBoosY** [comment]
Passed.

**Review by JackBoosY — APPROVED** [review]

**Comment by cenit** [comment]
May I ask you a comment about the fixes in #12199? IMHO they are very different and maybe more complete. These fixes do not solve almost anything related to the problems I had there in using assimp

**Comment by cenit** [comment]
Also another comment: irrlicht contains irrxml, but technically that port should use “external” irrxml (which I added and worked in the above PR)

**Comment by Neumann-A** [comment]
hmmm this also broke #13464. So why is irrxml embedded and not an extra port?

**Pull request merged**